src/main/java/seedu/address/model/folder/UniqueFolderList.java

```diff
@@ -37,6 +39,35 @@ public boolean contains(Folder toCheck) {
         return internalList.stream().anyMatch(toCheck::isSameFolder);
     }
 
+    /**
+     * Returns true if the list contains a folder with the same folder name.
+     */
+    public boolean containsFolderName(FolderName name) {
+        requireNonNull(name);
+        for (int i = 0; i < internalList.size(); i++) {
+            Folder folder = internalList.get(i);
+            if (folder.getFolderName().equals(name)) {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
In this class, there is already a method `contains()` written to check whether a list contains a folder. Perhaps you can use this one instead to prevent duplication of code?